File: ccc23s2.cpp
#include <iostream>

using namespace std;

int H[5000], S[5001];

int main() {
    int N;
    cin >> N;
    for (int i = 0; i < N; i++) {
        cin >> H[i];
        S[i + 1] = 5e+8;
    }
    for (int i = 0; i < N; i++) {
        int sym = 0;
        for (int l = i, r = i; l >= 0 && r < N; l--, r++) {
            int diff = r - l + 1;
            sym += abs(H[l] - H[r]);
            S[diff] = min(sym, S[diff]);
        }
        sym = 0;
        for (int l = i, r = i + 1; l >= 0 && r < N; l--, r++) {
            int diff = r - l + 1;
            sym += abs(H[l] - H[r]);
            S[diff] = min(sym, S[diff]);
        }
    }
    for (int i = 1; i < N; i++) cout << S[i] << " ";
    cout << S[N] << "\n";
    return 0;
}

File: ccc23s2.py
N = int(input())
h = []
h = input().split()
s = [2000000000] * (N + 1)

def check(l, r):
    sym = 0
    while l >= 0 and r < N:
        diff = r - l + 1
        sym += abs(h[l] - h[r])
        s[diff] = min(s[diff], sym)
        l -= 1
        r += 1

for i in range(n):
    check(i, i)
    check(i, i + 1)

print(*s[1:])